### PR TITLE
Presentation

### PR DESCRIPTION
**Pull request by bernardodiasc**
This PR address the following:
- #59 Profiles page
- #61 Header and main menu
- #63 Teams page
- #64 Profile page
- #67 Team page
- #69 Add and edit forms

Notes:
- Bootstrap 4 is included in Webpack task compiling in sass.
- Minimum possible of custom css is included aiming further theming to bootstrap variables mostly.
- Components css files uses the same naming than components js files, they are postcss, not sass.
- Markup resulted is valid HTML5 with proper outline.

Review:
- Some bugs noted in #92 to be addressed in other tasks.

[Review on Reviewable](https://reviewable.io/reviews/x-team/x-map/93)

### CODE REVIEW

**Comment by jedrzej** [comment]
Why do we need empty .gitignore files in client/css/components/form and client/css/components/pages?

File: client/js/app.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ import TeamAddPage from './components/pages/TeamAddPage';
 import App from './components/App';
 
 // Import the CSS file, which HtmlWebpackPlugin transfers to the build folder
```
Comment (jedrzej) [inline client/js/app.js:21]: Is this comment relevant to main.css or setup.scss?

File: client/js/components/forms/ProfileForm.js [hunk 2]
```diff
@@ -21,37 +22,63 @@ class ProfileForm extends Component {
     const { errors } = this.props;
 
     return (
-      <div>
+      <div id="ProfileForm">
```
Comment (jedrzej) [inline client/js/components/forms/ProfileForm.js]: Can you use class instead of id? I can imagine a situation when someone could use multiple ProfileForm component on a single page. We probably won't do it, but using ID here limits reusability.

File: client/js/components/forms/TeamForm.js [hunk 3]
```diff
@@ -19,29 +20,36 @@ class TeamForm extends Component {
...
     const { errors } = this.props;
-
-    let header = <h3>Add team</h3>;
-    if (this.state.id) {
-      header = <h3>Edit team #{this.state.id}</h3>;
-    }
-
     return (
-      <div>
+      <div id="TeamForm">
```
Comment (jedrzej) [inline client/js/components/forms/TeamForm.js:29]: See above - ProfileForm

File: client/js/components/forms/TeamForm.js [hunk 4]
```diff
@@ -19,29 +20,36 @@ class TeamForm extends Component {
...
-          </div>
-          <div className="row">
-            <button className="button" type="submit">Save</button>
-          </div>
+        <form onSubmit={this.onSubmit.bind(this)}>
+          <fieldset className="form-group">
+            <label>Team name*</label>
+            <input type="text" className="form-control" placeholder="Name" value={this.state.name}
+              minLength="2" maxLength="32" onChange={this.onInputChange.bind(this, 'name')} required/>
+            <small className="text-muted">Max length 32 chars.</small>
```
Comment (jedrzej) [inline client/js/components/forms/TeamForm.js]: Could you be a bit more elegant when communicating validation rules to the users? :) I think sth like "Maximum length: 32 characters"  or will sound much better

File: client/js/components/forms/TeamForm.js [hunk 5]
```diff
@@ -19,29 +20,36 @@ class TeamForm extends Component {
...
-            <textarea placeholder="Description" value={this.state.description} maxLength="1024" onChange={this.onInputChange.bind(this, 'description')}/>
-          </div>
-          <div className="row">
-            <button className="button" type="submit">Save</button>
-          </div>
+        <form onSubmit={this.onSubmit.bind(this)}>
+          <fieldset className="form-group">
+            <label>Team name*</label>
+            <input type="text" className="form-control" placeholder="Name" value={this.state.name}
+              minLength="2" maxLength="32" onChange={this.onInputChange.bind(this, 'name')} required/>
```
Comment (jedrzej) [inline client/js/components/forms/TeamForm.js]: Could you also change the min length in the backend? Line 25 in Team.php

Comment (jedrzej) [inline client/js/components/forms/TeamForm.js]: Or let's do it differently - once PR is ready, I'll add another commit that would update client and server validation rules so that they are the same

File: client/js/components/fragments/MiniProfile.js [hunk 6]
```diff
@@ -1,19 +1,32 @@
...
-
 class MiniProfile extends Component {
   render() {
     const { user } = this.props;
+
+    let avatar = null;
+    if (user.avatar) {
+      avatar = (
+        <span className="media-left">
+          <img className="media-object img-circle" src={user.avatar} alt={`${user.firstName} ${user.lastName}`}/>
```
Comment (jedrzej) [inline client/js/components/fragments/MiniProfile.js:12]: What happened to default avatar?

File: client/js/components/fragments/MiniProfile.js [hunk 7]
```diff
@@ -1,19 +1,32 @@
...
+        </span>
+      );
+    }
+
+    let name = (<p className="text-muted">{user.email}</p>);
```
Comment (jedrzej) [inline client/js/components/fragments/MiniProfile.js]: Single line elements don't need to be wrapped in () braces - I think we don't use braces anywhere else but this place

File: client/js/components/fragments/MiniProfile.js [hunk 8]
```diff
@@ -1,19 +1,32 @@
...
-        <Link to={`/profile/${user.id}`} title={'${user.firstName} ${user.lastName}'}>
-          <img src={user.avatar || DefaultAvatar} alt={'${user.firstName} ${user.lastName}'}/>
-          <p>{user.firstName} {user.lastName}</p>
-        </Link>
-      </span>
+      <Link className="miniProfile media" to={`/profile/${user.id}`} title={`${user.firstName} ${user.lastName}`}>
+          {avatar}
+        <span className="media-body">
+          <h4 className="media-heading">{user.firstName} {user.lastName}</h4>
+          {name}
```
Comment (jedrzej) [inline client/js/components/fragments/MiniProfile.js]: It's either email or slack ID so I wouldn't call it "name"  - I know, naming stuff is hard :). 
Could you also put `<p></p>` here and keep only content of the `<p>` in the variable? It's the same for both email and slack ID so no need to keep it in the variable.

So that in this line we have
`<p className="text-muted">{whateverYouDecideToCallTheVariable}</p>`

File: client/js/components/fragments/ProfileInformations.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+          <header href="#" className="panel-heading list-group-item-heading" role="tab" id="ProfileInformationsHeading">
+            <h4 className="panel-title" data-toggle="collapse" data-parent="#ProfileInformations"
+              aria-expanded="true" aria-controls="ProfileInformationsCollapse"
+              href="#ProfileInformationsCollapse">Informations...</h4>
+          </header>
+
+          <section id="ProfileInformationsCollapse" className="panel-collapse collapse in list-group-item-text"
+            role="tabpanel" aria-labelledby="ProfileInformationsHeading">
+            <div>
+              <label className="input-group">
```
Comment (jedrzej) [inline client/js/components/fragments/ProfileInformations.js]: Why is the class called "input-group" if there are no inputs here?

File: client/js/components/fragments/Team.js [hunk 10]
```diff
@@ -1,47 +1,16 @@
 import React, { Component, PropTypes } from 'react';
-import MiniProfile from './MiniProfile';
+import TeamInformations from './TeamInformations';
```
Comment (jedrzej) [inline client/js/components/fragments/Team.js]: See comments in ProfileInformations

File: client/js/components/pages/HomePage.js [hunk 11]
```diff
@@ -3,7 +3,7 @@ import DocumentTitle from 'react-document-title';
 
 class HomePage extends Component {
   render() {
-    return <DocumentTitle title="Homepage | X-Map"/>;
+    return <DocumentTitle title="X-Map | X-Team"/>;
```
Comment (jedrzej) [inline client/js/components/pages/HomePage.js:6]: This is inconsistent with titles on other pages - X-Map always goes second

File: client/js/components/pages/ProfilePage.js [hunk 12]
```diff
@@ -28,36 +28,48 @@ class ProfilePage extends Component {
       return <span/>;
     }
 
-    let editLink;
-    let setLocationLink;
+    // Use some better picture in this place
```
Comment (jedrzej) [inline client/js/components/pages/ProfilePage.js]: What does it mean? What better picture?

File: client/js/components/pages/TeamAddPage.js [hunk 13]
```diff
@@ -15,9 +15,16 @@ class TeamAddPage extends Component {
 
     return (
       <DocumentTitle title="Add team | X-Map">
-        <div className="panel">
-          <TeamForm onSubmit={actions.teamCreate} onSuccess={this.redirectToTeamsPage.bind(this)} errors={errors}/>
-        </div>
+        <article id="TeamAddPage" className="page card">
+          <header className="card-header">
+            <h3 className="card-title">Add team</h3>
+            <p className="card-subtitle">Create new team then edit the team to add people.</p>
```
Comment (jedrzej) [inline client/js/components/pages/TeamAddPage.js]: "Create a new team"
I was also thinking about adding users to teams on team page, not in the edit page, as link/unlink operation is a separate API request - it doesn't happen during team update. Doing that in team edit page would be a bit misleading as changes to team members would be persisted even if you don't press the save button

File: client/js/components/pages/TeamsPage.js [hunk 14]
```diff
@@ -38,26 +38,25 @@ class TeamsPage extends Component {
     let adminMenu = null;
     if (isAdmin) {
       adminMenu = (
-        <section className="navigation">
-          <Link to="/team/new" className="button">Add team</Link>
-        </section>
+        <div className="btn-group" role="group" aria-label="Actions menu">
+          <Link className="btn btn-secondary btn-sm" to="/team/new">Add team</Link>
```
Comment (jedrzej) [inline client/js/components/pages/TeamsPage.js:45]: I can see you added aria attributes all over the site + lots of titles/alt messages that I didn't remember to add, which is really good. But don't we need title on links as well?

File: client/package.json [hunk 15]
```diff
@@ -42,9 +45,12 @@
     "extract-text-webpack-plugin": "0.9.1",
     "file-loader": "^0.8.4",
     "html-webpack-plugin": "^1.6.1",
+    "imports-loader": "^0.6.5",
     "ip": "1.1.0",
+    "jquery": "^2.2.0",
```
Comment (jedrzej) [inline client/package.json:50]: Why do we need to import JQuery explicitely? Isn't it a dependency of some other package? I don't think we use jquery anywhere

**Comment by jedrzej** [comment]
Aren't glyphicons something we could install with npm instead of storing them in our source?

**Comment by jedrzej** [comment]
After build is done, favicon.png and blueMarker.png are corrupted

**Comment by jedrzej** [comment]
Could build process save images in build/img folder?

**Comment by jedrzej** [comment]
2 of the images we use are renamed to something that looks like hash of their content, the other 4 don't - why is that?

File: client/js/components/fragments/Map.js [hunk 16]
```diff
@@ -31,11 +31,10 @@ class Map extends Component {
...
 
   configureMap(google) {
-    this.map = new google.maps.Map(document.getElementById('map'), {
-      center: {lat: 0, lng: 50},
-      zoom: 2,
-      minZoom: 2,
-      mapTypeId: google.maps.MapTypeId.SATELLITE,
+    this.map = new google.maps.Map(document.getElementById('Map'), {
+      center: {lat: 0, lng: 0},
+      zoom: 3,
```
Comment (jedrzej) [inline client/js/components/fragments/Map.js]: Why did you remove the min zoom? When you zoom out to zoom=1 or zoom=1 the app looks bad

Comment (jedrzej) [inline client/js/components/fragments/Map.js]: I set zoom to 2 as then I could see the whole world by default on my laptop :) What screen are you using? With zoom=3 I can't see the whole world by default. Maybe we could somehow decide the default zoom by checking user's resolution or sth?

**Comment by jedrzej** [comment]
Cancel button on "Add team" screen leads to 404 page, it also gives " Cannot read property 'users' of undefined" error in the console

File: client/js/components/fragments/ProfileInformations.js [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+  render() {
+    const { user } = this.props;
+
+    return (
+      <div id="ProfileInformations" className="accordion list-group-item" role="tablist" aria-multiselectable="true">
+        <section className="panel panel-default">
+          <header href="#" className="panel-heading list-group-item-heading" role="tab" id="ProfileInformationsHeading">
+            <h4 className="panel-title" data-toggle="collapse" data-parent="#ProfileInformations"
+              aria-expanded="true" aria-controls="ProfileInformationsCollapse"
+              href="#ProfileInformationsCollapse">Informations...</h4>
```
Comment (jedrzej) [inline client/js/components/fragments/ProfileInformations.js]: Could it just say "Information" instead of "Information..."? I'd also change the header, linked teams is information as well. I'd call it User or General. Or skip it at all, as this page already says "Profile page" a few pixels above. The same for teams

**Comment by jedrzej** [comment]
Some general feedback:
- the loader screen looks much better without white border
- I liked the transparent header much more - it made buttons looked like there were additional map controls, they had more google-maps style
- user names in profile list and team names in team list are too big IMO
- field labels in profile/team page would look much better if they had the same width
- when rendering errors I'd show error message and the field name on the same line, without the BR in between
- Logout link is much wider than "My profile", it expands to the very end of the screen, is that on purpose?
- open-source and x-team links in the loader are blue, the background is blue as well which makes links not much visible

**Comment by jedrzej** [comment]
One small idea:
when I open the app, I just see map. When I click any of the links in the menu the box appears. I have to now go to the very left and click the logo to close the box. Could you add a small X button at the top right of the box? It would be a link to the homepage, but from user perspective it would let users kind of "close the box" and see the whole map again

File: client/js/app.js (hunk 1 above)
Comment (jedrzej) [inline client/js/app.js]: test

File: client/css/components/fragments/Header.css [hunk 18]
```diff
@@ -0,0 +1,20 @@
+#Header {
+  z-index: 9;
+  height: 56px;
+}
+
+#Header #logo h1 {
```
Comment (jedrzej) [inline client/css/components/fragments/Header.css]: Could we stay consistent in how we name elements? Header starts with upper case character, logo with lowercase character

**Comment by bernardodiasc** [comment]
Review status: 0 of 65 files reviewed at latest revision, 17 unresolved discussions.

---

_[client/css/components/fragments/Header.css, line 6 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC22-1-r1-6:-K9weFGCeJAx2vaIxIph:1113742438) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/css/components/fragments/Header.css#L6)):_
I consistently named all ID of unique components the same as the component itself. So the <Header> component in Header.js file uses #Header class on Header.css file (css components folder is mirrored from js components folder). The one is wrong in this standard is the #logo which can be a class. Please let me know if this is ok so I can change here? thanks

---

_[client/js/app.js, line 21 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21zW-r1-21:-K9wfZv3oeCQarjIIxO8:1388123757) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/app.js#L21)):_
this comment was there from the boilerplate, its valid for both files. setup.scss is needed for bootstrap that uses sass. our custom code is css (uses post css).

---

_[client/js/components/fragments/Map.js, line 36 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21zd-r1-36:-K9wg-1Y1DV4MuFcO_fs:-589556771) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/fragments/Map.js#L36)):_
maybe we should check based on viewport. on my big monitor zoom 3 cuts part of the map, and when I resize to small size, I need zoom 2 or 1 to see bigger part of the map. so this config can result differently depending on viewport size. I think we can set default to 2 (here will have a lot blank space above and below the map), remove the min zoom and check in other task about viewport specifics.

---

_[client/js/components/fragments/MiniProfile.js, line 12 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21ze-r1-12:-K9wgZoLHR0od6vEJ74x:-1465289288) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/fragments/MiniProfile.js#L12)):_
we can include, the aang avatar was just kidding, we need better image. I removed completely because I'm divided if "anonimous avatar" is better or worst than none. but I dont have preferences here.

---

_[client/js/components/fragments/ProfileInformations.js, line 19 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC22-C-r1-19:-K9whWJZlguUB51PnBhs:306330057) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/fragments/ProfileInformations.js#L19)):_
this is one part that can be better. I wanted the styles but bootstrap uses "input" in the naming and make stuff dont make sense, although is just styles and works as expected. I want to pimp this part where displays information a little bit in some other moment, if you agree.

---

_[client/js/components/pages/HomePage.js, line 6 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21zj-r1-6:-K9wiBR8IlCEI10ffWBW:-1065457193) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/pages/HomePage.js#L6)):_
this is the homepage, it's special case. :)

---

_[client/js/components/pages/ProfilePage.js, line 31 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21zm-r1-31:-K9wiLv4gllkwLSdCJFD:817788489) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/pages/ProfilePage.js#L31)):_
in the "card" display, this place will be very nice with some bigger image, can be a picture of the person or some landscape instead the avatar, avatar can be in other place in this card. Like what we have in x-team profiles, i.g. http://x-team.com/profile/bernardo-dias/

---

_[client/js/components/pages/TeamsPage.js, line 42 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21zs-r1-42:-K9wiyBKMS-V2yWwA9Cc:1729240796) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/pages/TeamsPage.js#L42)):_
good call. this accessibility enhancements is partially done. i'll make full review on this and microdata attrs. maybe we can skip this for this pr and handle in other task. the markup result in all pages is html5 valid already. :)

---

_[client/package.json, line 50 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21zu-r1-50:-K9wjX9MrcYG6Enq5IBU:99592614) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/package.json#L50)):_
we need this for bootstrap, I tried to leave jquery out without much success, so I included. currently only being used by accordion effect, but the bootstrap accordion effect is kinds broken, lots of glitches. if we could remove jquery and write custom js behaviours, would be much better. I'll not make changes on this in this PR b/c is time consuming. we can address in other issue.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93)_

**Comment by bernardodiasc** [comment]
> Why do we need empty .gitignore files in client/css/components/form and client/css/components/pages?

I mirrored the component folder in css from js, this this file is just to git track the folder. even not being used rigth now maybe is good to have as this standard suggests to use it in case custom styles for pages.

> Aren't glyphicons something we could install with npm instead of storing them in our source?

No sure, I've included bootstrap files, we can remove these files currently. not using any icon in the presentation so far.

> - After build is done, favicon.png and blueMarker.png are corrupted
> - 2 of the images we use are renamed to something that looks like hash of their content, the other 4 don't - why is that?

this is very strange... idk, I'll take a look.

> Could build process save images in build/img folder?

should be possible. not sure yet.

---

Review status: 0 of 65 files reviewed at latest revision, 17 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93#-:-K9wkzCrxVxpt1I_reTW:599300924)_

**Comment by bernardodiasc** [comment]
```
- the loader screen looks much better without white border
- I liked the transparent header much more - it made buttons looked like there were additional map controls, they had more google-maps style
- user names in profile list and team names in team list are too big IMO
- field labels in profile/team page would look much better if they had the same width
- when rendering errors I'd show error message and the field name on the same line, without the BR in between
- Logout link is much wider than "My profile", it expands to the very end of the screen, is that on purpose?
- open-source and x-team links in the loader are blue, the background is blue as well which makes links not much visible
```

got it. mostly visual changes are overrided by bootstrap defaults. with this move I didn't included any kind of styles that is not strict related to layout. even font-sizeand colors customization was left behind at this moment. I'll work on your feedback above as they are very simple and some is just to look what we had before, but a better work on styles/presentation to make a proper custom theme should be handled in other task. :)

---

Review status: 0 of 65 files reviewed at latest revision, 17 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93#-:-K9wmOZNiFM4zV6t5--O:1546281475)_

**Comment by bernardodiasc** [comment]
> Could you add a small X button at the top right of the box?

yes! I've think about that yesterday, very useful, good catch!

---

Review status: 0 of 65 files reviewed at latest revision, 17 unresolved discussions.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93#-:-K9wncnkZAqKMXGaCbfp:-41670103)_

**Comment by bernardodiasc** [comment]
Review status: 0 of 65 files reviewed at latest revision, 17 unresolved discussions.

---

_[client/js/components/fragments/ProfileInformations.js, line 13 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC22-C-r1-13:-K9x-U7deR2viwy5838s:705481251) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/fragments/ProfileInformations.js#L13)):_
This part is kind the way structure is built, the block items inside the page are wrapped the same accordion structure, this is the consistency we have to scale (include more blocks of whatever in each page). for humans can be redundant, but a crawler can identify as a section with the "informations" information inside, so if dont need to display, we can include `className="sr-only sr-only-focusable"` and remove aria attributes that bootstrap uses to apply the js behavior of accordeon.
I'll not change this now, just out of curiosity for next changes.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93)_

**Comment by bernardodiasc** [comment]
Review status: 0 of 65 files reviewed at latest revision, 17 unresolved discussions.

---

_[client/js/components/fragments/ProfileInformations.js, line 19 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC22-C-r1-19:-K9x0_pMhjmSkQ5J5qQ_:1627259253) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/fragments/ProfileInformations.js#L19)):_
not changing this now, but set fixed width in labels. I think it's looking nice for a while. also, we can consider inline editing for each field individualy instead the page edit. just an idea, can be discarded or included in backlog.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93)_

**Comment by jedrzej** [comment]
I'll create tickets for those broken images. I'll also start writing down ideas for UI improvements so that we can have a look at it later.

---

Reviewed 37 of 69 files at r1.
Review status: 24 of 59 files reviewed at latest revision, 13 unresolved discussions.

---

_[client/css/components/form/.gitignore, line 0 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w0Rwqbk1V3biPjXOK:-K9w0Rwqbk1V3biPjXOL:1290648178) ([raw file](https://github.com/x-team/x-map/blob/413aea9d643927fe978587312645c00f8e4be747/client/css/components/form/.gitignore#L0)):_
Why do we need empty .gitignore file here?

---

_[client/css/components/fragments/Header.css, line 6 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w0pHOoOHTNRBWiR_G:-K9w0pHPUk_ycrLTxy7K:-1828938755) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/css/components/fragments/Header.css#L6)):_
Could we stay consistent in how we name elements? Header starts with upper case character, logo with lowercase character

---

_[client/css/components/pages/.gitignore, line 0 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w1-t6q5UzxhcYFUqy:-K9w1-t6q5UzxhcYFUqz:1290648178) ([raw file](https://github.com/x-team/x-map/blob/413aea9d643927fe978587312645c00f8e4be747/client/css/components/pages/.gitignore#L0)):_
Why do we need empty .gitignore file here?

---

_[client/js/components/fragments/Map.js, line 36 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21zd-r1-36:-KA-PcYbnbu25fnWZjqM:-969486004) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/fragments/Map.js#L36)):_
I'll create a ticket for that

---

_[client/js/components/fragments/MiniProfile.js, line 12 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21ze-r1-12:-KA-PujnGUJ8GmTkSuaO:2065184899) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/fragments/MiniProfile.js#L12)):_
The reason I like default avatar is that layout does not break. Without that users with and without avatar are aligned differently.

---

_[client/fonts/glyphicons-halflings-regular.svg, line 1 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w1HILrZ5cTGMfqJwy:-K9w1HILrZ5cTGMfqJwz:1931810025) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/fonts/glyphicons-halflings-regular.svg#L1)):_
Aren't glyphicons something we could install with npm instead of storing them in our source?

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93#-:-KA-PNyNmCmSva-7NpEL:1163229045)_

**Comment by bernardodiasc** [comment]
Review status: 24 of 59 files reviewed at latest revision, 12 unresolved discussions.

---

_[client/css/components/fragments/Header.css, line 6 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w0pHOoOHTNRBWiR_G:-KA-cu5FYGlloyPpApOf:842492172) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/css/components/fragments/Header.css#L6)):_
seems my previous comment is hidden somewhere. the convention used is:
`/css/components` is mirrored folder/files structures of `/js/components`, Header.css stands for Header.js, and since it's unique component in the page it's using `#Header` id. All components stuff are using identical naming. Out of curiosity, while we use js and css in different folders, I've saw a lot setups putting files together (.js and .css of same components in the same place/folder). The reason is that the css in Header.css is strictly related to the component Header.js. Shared styles that uses classes (instead IDs, that is unique) are included in `/css/base` and `/css/layout`, normal css conventions there.

---

_[client/js/components/fragments/MiniProfile.js, line 12 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w-8ESHcOy0YtC21ze-r1-12:-KA-eVDFv5jJqXPKf7DZ:1531303572) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/js/components/fragments/MiniProfile.js#L12)):_
alright, let's include anonymous avatar. I'll create new ticket for that b/c without time now, maybe in the end of the day I can check this.

---

_[client/fonts/glyphicons-halflings-regular.svg, line 1 [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w1HILrZ5cTGMfqJwy:-KA-ek2jtzwBzeYkZSyJ:1413660444) ([raw file](https://github.com/x-team/x-map/blob/de5962cd99790763e18dfedb92c8ae306849f78a/client/fonts/glyphicons-halflings-regular.svg#L1)):_
I removed this already.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93)_

**Comment by bernardodiasc** [comment]
Review status: 24 of 59 files reviewed at latest revision, 12 unresolved discussions.

---

_[client/css/components/form/.gitignore, line  [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w0Rwqbk1V3biPjXOK:-KA-g6RmZJfHXrcO7hwW:-716696335) ([raw file](https://github.com/x-team/x-map/blob/413aea9d643927fe978587312645c00f8e4be747/client/css/components/form/.gitignore#L)):_
this file is jut for git to track the folder. wanted to keep this folder as `/css/components` mirror `/js/components/`, this file can be removed after include any other. (some people uses a file called .gitkeep, which means nothing too).

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93)_

**Comment by jedrzej** [comment]
Review status: 24 of 59 files reviewed at latest revision, 9 unresolved discussions.

---

_[client/css/components/form/.gitignore, line  [r1]](https://reviewable.io:443/reviews/x-team/x-map/93#-K9w0Rwqbk1V3biPjXOK:-KA-ggngEWu4t030CO84:1763986306) ([raw file](https://github.com/x-team/x-map/blob/413aea9d643927fe978587312645c00f8e4be747/client/css/components/form/.gitignore#L)):_
I know about .gitkeep - and indeed it does nothing. The problem I have with .gitignore is that it does not do nothing, as it's interpreted by git. So I'd switch to .gitkeep or just remove it completely, as we already have files in this folder. Up to you :)

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/x-team/x-map/93)_
